File: frontend/src/types/research.ts
export interface KeywordResult {
  keyword: string
  platform: string
  search_volume: number
  competition_level: string
  avg_price?: number | null
  total_results: number
  related_keywords: { keyword: string; volume: number; competition: string }[]
  trend_data: { date: string; volume: number }[]
}

export interface SavedResearch {
  id: string
  keyword: string
  platform: string
  search_volume?: number | null
  competition_level?: string | null
  avg_price?: number | null
  analyzed_at?: string | null
}

export interface Competitor {
  id: string
  platform: string
  name: string
  seller_name?: string | null
  price?: number | null
  sales_estimate?: number | null
  rating?: number | null
  review_count?: number | null
  is_tracked: boolean
  last_updated?: string | null
}

export interface TrendingProduct {
  id: string
  platform: string
  name: string
  price_min?: number | null
  price_max?: number | null
  sales_volume?: number | null
  sales_growth_rate?: number | null
  category_path?: string | null
  category_label?: string | null
  tags?: string[]
}

File: backend/app/schemas/research.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict, Field


class KeywordResearchResult(BaseModel):
    keyword: str
    platform: str
    search_volume: Optional[int] = None
    competition_level: Optional[str] = None
    avg_price: Optional[float] = None
    total_results: Optional[int] = None
    related_keywords: list[dict] = Field(default_factory=list)
    trend_data: list[dict] = Field(default_factory=list)


class SavedResearchCreate(BaseModel):
    keyword: str
    platform: str


class SavedResearchResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    keyword: str
    platform: str
    search_volume: Optional[int] = None
    competition_level: Optional[str] = None
    avg_price: Optional[float] = None
    analyzed_at: Optional[datetime] = None

class CompetitorCreate(BaseModel):
    platform: str
    platform_product_id: Optional[str] = None
    name: str
    seller_name: Optional[str] = None
    price: Optional[float] = None
    url: Optional[str] = None


class CompetitorResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    platform: str
    name: str
    seller_name: Optional[str] = None
    price: Optional[float] = None
    sales_estimate: Optional[int] = None
    rating: Optional[float] = None
    review_count: Optional[int] = None
    is_tracked: bool = True
    last_updated: Optional[datetime] = None

class TrendingProductResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    platform: str
    name: str
    price_min: Optional[float] = None
    price_max: Optional[float] = None
    sales_volume: Optional[int] = None
    sales_growth_rate: Optional[float] = None
    category_path: Optional[str] = None
    category_label: Optional[str] = None  # Resolved Chinese label from dictionary
    tags: list[str] = Field(default_factory=list)

class TrendingProductCreate(BaseModel):
    platform: str
    name: str
    price_min: Optional[float] = None
    price_max: Optional[float] = None
    sales_volume: Optional[int] = None
    sales_growth_rate: Optional[float] = None
    category_path: Optional[str] = None
    market: Optional[str] = None  # Market tag for filtering (MY/PH/SG/TH/VN/ID)


class SyncStats(BaseModel):
    shopee: int = 0
    temu: int = 0
    tiktok: int = 0
    total: int = 0
    errors: list[str] = Field(default_factory=list)
